packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,465 @@ class PositionedDirectional extends StatelessWidget {
+/// you'll see an "overflow error" (during development you will see black and yellow
+/// stripes and an exception, but a production app just clips the content). For content
+/// that should wrap to a new line, use [Wrap]. For scrolling content
+/// prefer [SliverList] over [ListView] or [SingleChildScrollView], for
+/// better performance.
+///
+/// * **Performance:** It is recommended to check your app's
+/// performance using the Flutter DevTools, but make sure to do this in
+/// "profile mode" for accurate results. Development builds will always
+/// seem slow.
+///
+/// ## Glossary of Terms:
+///
+/// For reference, here are several terms used below to describe the layout algorithm:
+///
+/// [BoxConstraints] : These are the minimum and maximum sizes which a parent
+/// [Widget] dictates its child must be. It can be said a child is tightly
+/// constrained or loosely constrained. "Tightly
+/// constrained" means the size of the child is tightly controlled by its
+/// parent [Widget], and this is achieved by setting the same value for
+/// both the minimum and maximum sizes. With tight constraints the size is fixed, as it
+/// cannot be more or less than the value set for minimum and maximum. If the
+/// minimum and maximum values are not the same, it is said the child has
+/// "loose constraints", and so can be any size (in that axis) as long as it
+/// is between those values. The constraints for both height and width are
+/// found within the [BoxConstraints].
+///
+/// [Flex] : The superclass of [Row] and [Column]. All [Row]s and [Column]s
+/// are [Flex]es, and for most of this explanation they will be referred
+/// to as a [Flex].
+///
+/// [Flexible.flex] : A property of a [Flexible] that should not be
+/// confused with a [Flex]. It determines how much "strength" (in
+/// some frameworks this is called "gravity") a [Flexible] has compared
+/// to other [Flexible]s. More on this in step two of the layout algorithm,
+/// below.
+///
+/// [MainAxis]: The main direction of the [Flex]. This is vertical
+/// for a [Column] and horizontal for a [Row], and corresponds to the
+/// [direction] property of a [Flex].
+///
+/// [CrossAxis]: The other direction. The cross axis of a [Column]
+/// is horizontal, and the cross axis of a [Row] is vertical. It is
+/// always perpendicular to the [MainAxis].
+///
+/// [Flexible] : A widget that is a child of a [Flex]. It _could_ cause
+/// its child to stretch in the direction of the main axis, _though it
+/// does necessarily have to._ It cannot be a child of anything other
+/// than a [Flex].
+///
+/// [Expanded] : A child of a [Flex], it is a subtype of [Flexible]
+/// that _must_ stretch its child to use all of the space _reserved for
+/// it_ during layout. An [Expanded] will always use all of the space it is
+///  offered but this does not guarantee it will use all of the extra space
+///  in the [Flex] because it may not be offered all of the extra space
+///  within the [Flex].
```

Comment:
The fact that it might not use all the space is one of the most frustrating things to do with `Expanded` that people run into, and I think this is a good explanation of it.